nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/osgi/EmbeddedOSGiGlassFishRuntime.java

```diff
@@ -144,20 +156,41 @@ public synchronized GlassFish newGlassFish(GlassFishProperties gfProps) throws G
 
     @Override
     public synchronized void shutdown() throws GlassFishException {
-        // make a copy to avoid ConcurrentModificationException
-        for (GlassFish gf : new ArrayList<>(gfs)) {
-            if (gf.getStatus() != GlassFish.Status.DISPOSED) {
+        // Make a copy to avoid ConcurrentModificationException
+        for (GlassFish glassFish : new ArrayList<>(glassFishes)) {
+            if (glassFish.getStatus() != DISPOSED) {
                 try {
-                    gf.dispose();
+                    glassFish.dispose();
                 } catch (GlassFishException e) {
                     e.printStackTrace();
                 }
             }
         }
-        gfs.clear();
+
+        glassFishes.clear();
         shutdownInternal();
     }
 
+    private MultiException findMultiException(Throwable t) {
+        Throwable currentThrowable = t;
+
+        while (currentThrowable != null) {
+            if (currentThrowable instanceof MultiException) {
+                return (MultiException) currentThrowable;
+            }
+
+            for (Throwable suppressed : currentThrowable.getSuppressed()) {
+                MultiException multiException = findMultiException(suppressed);
+                if (multiException != null) {
+                    return multiException;
```

Comment:
But this way you lose other possible suppressed exceptions describing what happened, right?
I need some more detailed explanation (javadoc), at least why do you think it is ok. And comparison of the output to see it too.

Reply:
> But this way you lose other possible suppressed exceptions describing what happened

I don't think so, as the original exception (throwable t) is still propagated:

```
throw new GlassFishException(bundleBuilder.toString(), t);
```

The first `MultiException` encountered is just used for `bundleBuilder.toString`.

Reply:
Oh, I see, thanks!